src/ImageSharp/Formats/WebP/WebPLossyDecoder.cs

```diff
@@ -0,0 +1,1414 @@
+            int dqy1Dc = hasValue ? this.bitReader.ReadSignedValue(4) : 0;
+            hasValue = this.bitReader.ReadBool();
+            int dqy2Dc = hasValue ? this.bitReader.ReadSignedValue(4) : 0;
+            hasValue = this.bitReader.ReadBool();
+            int dqy2Ac = hasValue ? this.bitReader.ReadSignedValue(4) : 0;
+            hasValue = this.bitReader.ReadBool();
+            int dquvDc = hasValue ? this.bitReader.ReadSignedValue(4) : 0;
+            hasValue = this.bitReader.ReadBool();
+            int dquvAc = hasValue ? this.bitReader.ReadSignedValue(4) : 0;
+            for (int i = 0; i < WebPConstants.NumMbSegments; ++i)
+            {
+                int q;
+                if (vp8SegmentHeader.UseSegment)
+                {
+                    q = vp8SegmentHeader.Quantizer[i];
+                    if (!vp8SegmentHeader.Delta)
+                    {
+                        q += baseQ0;
+                    }
+                }
+                else
+                {
+                    if (i > 0)
+                    {
+                        decoder.DeQuantMatrices[i] = decoder.DeQuantMatrices[0];
+                        continue;
+                    }
+                    else
+                    {
+                        q = baseQ0;
+                    }
+                }
+
+                Vp8QuantMatrix m = decoder.DeQuantMatrices[i];
+                m.Y1Mat[0] = WebPLookupTables.DcTable[Clip(q + dqy1Dc, 127)];
+                m.Y1Mat[1] = WebPLookupTables.AcTable[Clip(q + 0, 127)];
+                m.Y2Mat[0] = WebPLookupTables.DcTable[Clip(q + dqy2Dc, 127)] * 2;
+
+                // For all x in [0..284], x*155/100 is bitwise equal to (x*101581) >> 16.
+                // The smallest precision for that is '(x*6349) >> 12' but 16 is a good word size.
+                m.Y2Mat[1] = (WebPLookupTables.AcTable[Clip(q + dqy2Ac, 127)] * 101581) >> 16;
+                if (m.Y2Mat[1] < 8)
+                {
+                    m.Y2Mat[1] = 8;
+                }
+
+                m.UvMat[0] = WebPLookupTables.DcTable[Clip(q + dquvDc, 117)];
+                m.UvMat[1] = WebPLookupTables.AcTable[Clip(q + dquvAc, 127)];
+
+                // For dithering strength evaluation.
+                m.UvQuant = q + dquvAc;
+            }
+        }
+
+        private void ParseProbabilities(Vp8Decoder dec)
+        {
+            Vp8Proba proba = dec.Probabilities;
+
+            for (int t = 0; t < WebPConstants.NumTypes; ++t)
+            {
+                for (int b = 0; b < WebPConstants.NumBands; ++b)
+                {
+                    for (int c = 0; c < WebPConstants.NumCtx; ++c)
+                    {
+                        for (int p = 0; p < WebPConstants.NumProbas; ++p)
+                        {
+                            byte prob = WebPLookupTables.CoeffsUpdateProba[t, b, c, p];
+                            int v = this.bitReader.GetBit(prob) != 0
```

Comment:
let v be a byte variable, cast the ReadValue() case to byte and remove the - the redundant - cast in line 1298.